src/ast/codegen_llvm.cpp

```diff
@@ -991,7 +991,12 @@ void CodegenLLVM::visit(FieldAccess &acc)
     // The struct we are reading from has not been pulled into BPF-memory,
     // so expr_ will contain an external pointer to the start of the struct
 
-    Value *src = b_.CreateAdd(expr_, b_.getInt64(field.offset));
+    // LLVM doesn't like Binops with different types, so we convert our pointer
+    // to integer and then back to pointer.
+    // Value *src = b_.CreateAdd(expr_, b_.getInt64(field.offset));
```

Comment:
Maybe to make it more clear for future readers:
```suggestion
    // Instead of doing:
    //     Value *src = b_.CreateAdd(expr_, b_.getInt64(field.offset));
    // we do the below:
```